src/workerd/util/sqlite.c++

```diff
@@ -257,6 +258,31 @@ SqliteDatabase::Query::ValuePtr SqliteDatabase::Query::getValue(uint column) {
   KJ_UNREACHABLE;
 }
 
+SqliteDatabase::Query::ValueOwned SqliteDatabase::Query::getValueOwned(uint column) {
+  switch (sqlite3_column_type(statement, column)) {
+    case SQLITE_INTEGER: {
+      int64_t value = getInt64(column);
+      constexpr uint64_t MAX_SAFE_INTEGER = (1ull << 53) - 1;
+      // We could return a BigInt (int64_t) if the value does not fit within MAX_SAFE_INTEGER,
+      // however this would result in confusing behavior to change the returned value based
+      // on how large it is. To avoid this, we explicitly require the user / sql statement
+      // to convert larger values to a string first.
+      JSG_REQUIRE(abs(value) <= MAX_SAFE_INTEGER, Error,
+        "Integers larger than MAX_SAFE_INTEGER must be converted to string, value: ", value);
+      return static_cast<double>(value);
+    }
+    case SQLITE_FLOAT:   return getDouble(column);
+    case SQLITE_TEXT:    return kj::heapString(getText(column));
```

Comment:
is `SQLLITE_TEXT` always expected to be UTF-8?

Reply:
SQLite automatically converts text to be in the requested encoding if it isn't already, and we always request UTF-8, yes.

Reply:
Cool ok, just wanted to double ✔️